src/app/georeferencer/qgsvectorwarper.cpp

```diff
@@ -0,0 +1,57 @@
+/***************************************************************************
+     qgsvectorwarper.cpp
+     --------------------------------------
+    Date                 :
+    Copyright            :
+    Email                :
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include <cmath>
+#include <cstdio>
+
+#include <cpl_conv.h>
+#include <cpl_string.h>
+#include <gdal.h>
+#include <ogr_spatialref.h>
+
+#include <QFile>
+#include <QProgressDialog>
+
+#include "qgsvectorwarper.h"
+#include "qgsrunprocess.h"
+
+
+
+
+QgsVectorWarper::QgsVectorWarper( QWidget *parent )
+  : mParent( parent )
+{
+}
+
+bool QgsVectorWarper::executeGDALCommand( const QString &fused_command )
+{
+    QString command;
+    QStringList arguments;
+    command, arguments = QgsRunProcess::splitCommand( fused_command );
+    if ( arguments.length() == 1 )
+        arguments.replace( 0 ,arguments.first().replace( "ogr2ogr ","") );
+    else if ( arguments.first() == QString( "ogr2ogr" ) )
+        arguments.removeFirst();
+    if ( command.isEmpty() )
+        command = QString( "ogr2ogr" );
+
+    QgsBlockingProcess *proc = new QgsBlockingProcess( command, arguments );
+    //proc.setStdOutHandler(on_stdout);
+    //proc.setStdErrHandler(on_stderr);
+    int res = proc->run();
+    if ( proc->exitStatus() == QProcess::NormalExit )
```

Comment:
@nyalldawson @rouault I tried to use qgsblockingprocess to call 'ogr2ogr' with the arguments, but I get an invalid process error. It seems that the system cannot get 'ogr2ogr'. Now should I add and tweaks all the system variables as is done in processing or is there a more reliable way to call 'ogr2ogr'? I have looked at the gdal api but I'm uncertain about the proper way to doing the call with all the needed changes (gcp, transforms, change of projection maybe, and possibly change of file type.)

Reply:
Can you leave this part temporarily? By lucky coincidence, I've a contract with NRCan to develop a plugin which requires gcp based transformation of geometries on a vertex-by-vertex basis. I'll make a QGIS wrapper around the underlying GDAL api, so that we can directly apply a gcp based transform to a QgsGeometry. Then this part of the georeferencer will become a simple matter of iterating through all the features in the layer and transforming their geometries using the new class.

This api should be inbound soon...! :+1:

Reply:
@nyalldawson Glad to hear NRCan is funding you!